server/src/toggle.rs

```diff
@@ -71,25 +96,40 @@ pub(super) async fn toggle(
     Path((eid, secret, qid, property)): Path<(Uuid, String, Uuid, Property)>,
     State(dynamo): State<Backend>,
     body: String,
-) -> Result<(), StatusCode> {
+) -> Result<Json<serde_json::Value>, StatusCode> {
     super::check_secret(&dynamo, &eid, &secret).await?;
 
-    let set = match &*body {
-        "on" => true,
-        "off" => false,
+    let req = match (&*body, property) {
+        ("on", Property::Hidden) => ToggleRequest::Hidden(true),
+        ("off", Property::Hidden) => ToggleRequest::Hidden(false),
+        ("on", Property::Answered) => ToggleRequest::Answered(Some(SystemTime::now())),
+        ("off", Property::Answered) => ToggleRequest::Answered(None),
         _ => {
             error!(%qid, body, "invalid toggle value");
             return Err(http::StatusCode::BAD_REQUEST);
         }
     };
 
-    match dynamo.toggle(&qid, property, set).await {
+    match dynamo.toggle(&qid, req).await {
         Ok(_) => {
             debug!(%eid, %qid, p = ?property, "toggled question property");
-            Ok(())
+            match req {
+                ToggleRequest::Hidden(set) => Ok(Json(serde_json::json!({ "hidden": set }))),
+                ToggleRequest::Answered(time) => {
+                    if let Some(time) = time {
+                        Ok(Json(serde_json::json!({
+                            "answered": get_dynamo_timestamp(time)
+                                .as_n().ok()
+                                .and_then(|v| v.parse::<usize>().ok())
```

Comment:
This feels weird — why go via `AttributeValue` at all here? `time` is already a `SystemTime`, so you can just take `duration_since(UNIX_EPOCH)` and it'll give you the right value.

Reply:
Good catch, not sure what I was thinking when writing this. I switched to just using SystemTime for this call.